src/hooks/useMedia.ts

```diff
@@ -1,14 +1,24 @@
 import { UseBaseQueryResult, useQuery } from 'react-query';
 
 import { getMediaById } from '../services/api.service';
+import { transformMediaItem } from '../utils/media';
 
 import type { PlaylistItem } from '#types/playlist';
 
 export type UseMediaResult<TData = PlaylistItem, TError = unknown> = UseBaseQueryResult<TData, TError>;
 
 export default function useMedia(mediaId: string, enabled: boolean = true): UseMediaResult {
-  return useQuery(['media', mediaId], () => getMediaById(mediaId), {
-    enabled: !!mediaId && enabled,
-    keepPreviousData: true,
-  });
+  return useQuery(
+    ['media', mediaId],
+    async () => {
+      const media = await getMediaById(mediaId);
```

Comment:
transformMediaItem can go into the .then inside getMediaById

Reply:
I've moved the transformers to the API service instead